Forte.EpiServer.AzureSearch/Extensions/ContentDataExtensions.cs

```diff
@@ -0,0 +1,18 @@
+using EPiServer.Core;
+using EPiServer.ServiceLocation;
+using Forte.EpiServer.AzureSearch.ContentExtractor.Block;
+
+namespace Forte.EpiServer.AzureSearch.Extensions
+{
+    public static class ContentDataExtensions
+    {
+        public static string ExtractTextFromBlock(this IContentData blockContent)
+        {
+            const string blockExtractedTextSeparator = " ";
+            var blockExtractorsController = ServiceLocator.Current.GetInstance<IBlockContentExtractorController>();
```

Comment:
I'm not a fan of extension methods which hides external dependencies. Can we introduce separate class which will have ExtractTextFromBlock method and will have IBlockContentExtractor controller injected via regular dependency injection? (and this class will be injected into all places where extension method is used)

Reply:
I tried using it like that, but it will cause a circular dependency error during initialization.
"IBlockContentExtractor controller "-> DI in ctor -> "IEnumerable<IBlockContentExtractors>" -> resolves to "IndexableContentExtractor" (among others) or any other ContentExtractor for nested blocks resolving and it will require IBlockContentExtractor, which closes the circle.